test/alphabet/alphabet_test.cpp

```diff
@@ -154,8 +154,9 @@ TYPED_TEST(alphabet, swap)
 
 TYPED_TEST(alphabet, assign_char)
 {
+    using char_t = typename TypeParam::char_type;
     TypeParam t0;
-    for (char i = std::numeric_limits<char>::min(); i < std::numeric_limits<char>::max(); ++i)
+    for (char_t i = std::numeric_limits<char_t>::min(); i < std::numeric_limits<char_t>::max(); ++i)
```

Comment:
> Is there an unknown value in the alphabet that we can set the value to in case an unknown/invalid character is passed to assign_char?

The behaviour for alphabets without wild card is to use the lexicographically smallest character in that case. But the quality alphabets will change a little anyway: #39